Fixing issues in Latin syllabification

dstelzer opened the pull request:
This fixes issue #993 (incorrect syllabification of _sumptus_ and similar words), and adds a flag to the Syllabifier object to turn off automatic i-to-j conversion—e.g. if your text is already marked with J's, and you want the syllabifier to take that into account. Previously, _Iulus_ would syllabify as _ju-lus_; now, you can choose if you want _ju-lus_ or _i-u-lus_.

(This is my first pull request, so please tell me if I've submitted it wrong.)

kylepjohnson commented:
Daniel, looks like you broke one test: https://travis-ci.org/github/cltk/cltk/jobs/717769154#L2220 . *If you need to change the test*, to reflect your new algo updates, please flag them in the comments here. (You'll see some other errors in that Travis build, concerning Old English -- you can ignore those.)

To run doctests only on this module, the command (IIRC) is: `nosetests --no-skip --with-doctest cltk/prosody/latin/syllabifier.py`

Please reach out if you get stuck :)

dstelzer commented:
(Stupid mistake on my part this time—I ran the tests on my end before pushing, but incorrectly, so I missed some of the scanner errors. Mea culpa.)

kylepjohnson commented:
No problem. Ping me here when you think it's ready.

kylepjohnson commented:
Or is this it? If so, I will merge tomorrow once the build is complete!

dstelzer commented:
There we go, I believe that fixes everything! The only errors in Travis now are from Old English; I adjusted the Latin tests to account for the new syllabifier behavior, and added some new ones on top of that.

Thanks for your patience; this is my first real pull request and I'm still somewhat finding my feet.

@kylepjohnson

kylepjohnson commented on cltk/prosody/latin/syllabifier.py:
```diff
@@ -87,7 +88,11 @@ def syllabify(self, words: str) -> List[str]:
         ['pul', 'cher']
         >>> print(syllabifier.syllabify("ruptus"))
-        ['ru', 'ptus']
+        ['rup', 'tus']
```
Agreed, this looks better.

kylepjohnson commented:
Thanks this looks good. The Old English doctests need to be fixed before the build server pushes to PyPI. I will open a ticket to this and see how much work it'll be.

(Note: we are focusing on a new major release, which has caused some repos to break.)
